sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -1931,18 +1931,29 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         sql(
           """
             |SELECT c1, s, s * 10 FROM (
-            |  SELECT c1, (SELECT FIRST(c2) FROM t2 WHERE t1.c1 = t2.c1) s FROM t1)
+            |  SELECT c1, (SELECT MIN(c2) FROM t2 WHERE t1.c1 = t2.c1) s FROM t1)
```

Comment:
what's the error if we don't make this change?

Reply:
Just a side note - I have been arguing, that first/last should be deterministic functions, but it has not gotten any attention - #29810.

Reply:
Can we not change the test query and assert the error instead? 

> Just a side note - I have been arguing, that first/last should be deterministic functions

+1 even though FIRST/LAST are not truly deterministic during execution. 

The purpose of this field is for determining the eligibility of query rewrites. Postgres has a nice categorization of those:
https://www.postgresql.org/docs/8.3/xfunc-volatility.html

SUM, AVG are not completely deterministic (when running distributed-ly) neither, but we can still do query optimizations over them, and I think it'd be fine for LAST/FIRST belong too.